### PR TITLE
Add PrivateKeySecretManager

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/client/secret/private_key.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    async fn generate_evm_addresses(
+        &self,
+        _coin_type: u32,
+        _account_index: u32,
+        _address_indexes: Range<u32>,
+        _options: impl Into<Option<GenerateAddressOptions>> + Send,
+    ) -> Result<Vec<EvmAddress>, Self::Error> {
+        panic!()
```
Comment (thibault-martinez) [inline sdk/src/client/secret/private_key.rs]: There is no proper Error I can use and the enum is not non_exhaustive so I can't add a new variant, not sure what to do...

Comment (Thoralf-M) [reply sdk/src/client/secret/private_key.rs]: What about `Err(Error::UnsupportedOperation.into())`?

Comment (Thoralf-M) [reply sdk/src/client/secret/private_key.rs]: Ok that's behind the ledger nano feature flag, but what about `SecretManagerMismatch`? Not exactly the same, but better than panic

File: sdk/src/client/secret/private_key.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        super::default_sign_transaction(self, prepared_transaction_data).await
+    }
+}
+
+impl PrivateKeySecretManager {
+    /// Create a new [`PrivateKeySecretManager`] from a base 58 encoded private key.
+    pub fn try_from_b58<T: AsRef<[u8]>>(b58: T) -> Result<Self, Error> {
+        let mut bytes = [0u8; ed25519::SecretKey::LENGTH];
+
+        if bs58::decode(b58.as_ref()).onto(&mut bytes).unwrap() != ed25519::SecretKey::LENGTH {
```
Comment (Thoralf-M) [inline sdk/src/client/secret/private_key.rs]: Here we could maybe use `iota_crypto::Error::PrivateKeyError`

File: sdk/tests/client/secret_manager/mod.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2022 IOTA Stiftung
```
Comment (Thoralf-M) [inline sdk/tests/client/secret_manager/mod.rs]:
```suggestion
// Copyright 2023 IOTA Stiftung
```

File: sdk/src/client/secret/private_key.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/client/secret/private_key.rs]: Changed

File: sdk/src/client/secret/private_key.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/client/secret/private_key.rs]: Changed

File: sdk/src/client/secret/private_key.rs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (Thoralf-M) [inline sdk/src/client/secret/private_key.rs]: The unwrap is safe?

Comment (abdulmth) [inline sdk/src/client/secret/private_key.rs]: can't we return the same error on the `unwrap`? can't an invalid character cause a panic? 🤔

Comment (thibault-martinez) [reply sdk/src/client/secret/private_key.rs]: Forgot about it, changed

Comment (thibault-martinez) [reply sdk/src/client/secret/private_key.rs]: Done

**Review by abdulmth — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Pull request merged**